File: 2283-SortEvenAndOddIndicesIndependently/2283-SortEvenAndOddIndicesIndependently.js
// Last updated: 3/6/2026, 2:08:41 PM
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortEvenOdd = function(nums) {
    let odd = [];
    let even = [];
    let res = [];

    for(let i=0;i<nums.length;i++){
        if(i%2==0){
            even.push(nums[i]);
        }
        else{
            odd.push(nums[i]);
        }
    }

    even = even.sort((a,b)=> a-b);
    odd = odd.sort((a,b)=> b-a);
    let j=0;k=0;
    for(let i=0;i<nums.length;i++){
        if(i%2==0){
            res.push(even[j])
            j++;
        }
        else{
            res.push(odd[k]);
            k++;
        }
    }
    return res;
};

File: 2283-SortEvenAndOddIndicesIndependently/2283-SortEvenAndOddIndicesIndependently.cpp
// Last updated: 7/17/2026, 4:25:33 PM
class Solution {
public:
    vector<int> sortEvenOdd(vector<int>& nums) {
        vector<int> even, odd, res;

        for (int i = 0; i < nums.size(); i++) {
            if (i % 2 == 0) {
                even.push_back(nums[i]);
            } else {
                odd.push_back(nums[i]);
            }
        }

        sort(even.begin(), even.end()); 
        sort(odd.begin(), odd.end(), greater<int>()); 

        int j = 0, k = 0;

        for (int i = 0; i < nums.size(); i++) {
            if (i % 2 == 0) {
                res.push_back(even[j++]);
            } else {
                res.push_back(odd[k++]);
            }
        }

        return res;
    }
};